### ISSUE TITLE
oauth_signature in authorizationHeaderForMethod: parameters error when add body

### ISSUE THREAD

**Issue by gali8** [AFNetworking/AFOAuth1Client#67]
I'm using AFOAuth1Client fore read and create products in Magento.

Well, i have a NSDictionary of product properties; This is the body of the request.

I'm calling the method:

``` objective-c
- (void)postPath:(NSString *)path
      parameters:(NSDictionary *)parameters
         success:(void (^)(AFHTTPRequestOperation *operation, id responseObject))success
         failure:(void (^)(AFHTTPRequestOperation *operation, NSError *error))failure
```

where the parameters argument is the dictionary above.

When AFOAuth1Client execute the method

```
- (NSString *)authorizationHeaderForMethod:(NSString*)method
                                      path:(NSString*)path
                                parameters:(NSDictionary *)parameters
```

it sets the `oauth_signature` calling the method `OAuthSignatureForMethod...` like below.

``` objective-c
    [mutableAuthorizationParameters setValue:[self OAuthSignatureForMethod:method path:path parameters:mutableParameters token:self.accessToken] forKey:@"oauth_signature"];
```

The issue is that the mutableParameters object, passed like "parameters", is not correct (it contains the `oauth_` and the products key/value properties).

If you replace `mutableParameters` with `mutableAuthorizationParameters` (`mutableAuthorizationParameters` contains only the `oauth_` key/value properties) it will works perfectly.

**Comment by mattt** [comment]
This should be fixed as of 80dd78c65a208b7bb8c27c74a8c8a347df650334.

**Comment by gali8** [comment]
Great job Mattt!
You rock! :)

Daniele from g8production.com
